src/components/TokenSymbol/index.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { setImageToPlaceholder } from 'src/routes/safe/components/Balances/utils'
+import Img from '../layout/Img'
+import { useToken } from 'src/logic/tokens/hooks/useToken'
+
+type Props = {
+  tokenAddress: string
+  height?: number
+}
+
+export const TokenSymbol = ({ tokenAddress, height = 26 }: Props): React.ReactElement => {
+  const token = useToken(tokenAddress)
```

Comment:
we should not use any logic inside `src/components` folder. can you remove the useToken from here? 

Also, would be great if you can create a story for this component. We started doing stories with the new layout, you could take an example for there, it should be pretty easy.

Reply:
> we should not use any logic inside src/components folder. can you remove the useToken from here? 

why is that?

Reply:
Because it's easier to understand and test them.

Reply:
What do you suggest?

Reply:
@Agupane I just realized that we possibly don't need the component at all and we can be fine just using a hook. What do you think?

Reply:
I prefer to wrap it in a component to we can avoid the <img> thing logic, maybe we can send to it the token as props so we can remove the logic and use the hook outside, on the place where <TokenSymbol> is called, something like:

```
const tokenInfo = useToken(tokenAddress)

return (
...
<TokenSymbol token={tokenInfo}/>
)
```

what do you think?

Reply:
What is the logic you want to avoid? Didn't get it, sorry

Reply:
This one:
`  const publicResult = await bucket.makePublic();
  console.log("bucket acl", publicResult);`

That's what I understood from @nicosampler , please confirm :)

Reply:
This is not connected to the comment, I mean in general that in places where we use the component we can use this hook instead

Reply:
Yes we can but we would need to copy and paste this logic in every place:
`<Img alt={token.name} height={height} onError={setImageToPlaceholder} src={imgUrl} />`

I think having a <TokenSymbol token={token} /> component looks more transparent

Reply:
But isn't it the same? You import a component and pass the address there vs you import the hook and pass the address there

Reply:
In the example above it would be
```const token = useToken(address)```
And that's it, don't see any complexity/overhead there

Reply:
Yes the result it's the same and it's not less complex, but looks cleaner IMO, anyways, I'll remove it and implement it with `<Img alt={token.name}..../>` :)

Reply:
Sorry I didn't get it. What is your plan now?

Also this component seems to be misleading, the name is TokenSymbol but it renders an image

Reply:
Yes, @Agupane that's right, `useToken` should be provided as a prop from outside.

> I don't understand how the ease of testing and understanding is connected with where the component is located.

@mikheevm  It's not where they are located, it's what they consume. If we have dumb components, thy should not consume any logic directly. I think that's the best way to organize our code, but perhaps you have a different idea.

Reply:
> useToken should be provided as a prop from outside.

passing a hook via props? why?

Reply:
what? no, no the hook the value of the hook.

Reply:
Let me jump into the discussion and try to understand.

Currently, we have a component `<TokenSymbol />` that receives a `tokenAddress` prop. With that info, it queries the store looking for _token information_ to display its content.

@nicosampler, is suggesting to make the component simpler and unaware of a store, so we should pass the props that the component requires, that would be, in my opinion, something around:

```typescript
const tokenInfo = useToken(tokenAddress)
return <TokenSymbol {...tokenInfo} />
```

What @Agupane purposed was something similar but passing an object instead. It's almost the same, one is tied to a structure, the other is a bit more dynamic because you can see which props are required or not.

With this, you're able to test a simple component by passing props at will and not depending on a store.

In regard to the name, as @mikheevm said, yes, it should be different as it's displaying the token's logo/image.

Reply:
@fernandomg <TokenSymbol /> is just a text, we don't need a separate component for that
```
const { tokenSymbol } = useToken(tokenAddress)

<Text>{tokenSymbol}</Text>
```
and that's it

same goes for a token logo, balance or whatsoever

Reply:
Ok, but abstraction for a repeated pattern is desirable...

So, better we move away from `components` this component in particular.

We pass the `tokenAddress` and let the component gather all the required information.

So we can have `<TokenLogo address={tokenAddress} />`, that simplifies the `<Img />` thing.

Reply:
I agree that we may want to have TokenLogo component because it has other logic in place (like setting the image to placeholder). But I don't understand why we want a component for TokenSymbol, this is just plain text.

Reply:
oh, I agreed with you that the name is wrong. This component displays the logo.

To sum up:
1. rename `TokenSymbol` to `TokenLogo`
2. move `TokenLogo` from `src/components/` to where??

Reply:
I would make it receive the logo link as a prop and keep it there

Reply:
Being that specific, you're moving

```javascript
const imgUrl = 'image' in token ? token.image : token.logoUri
```

...outside the component and the idea is that it should simplify the things.

But I agree that we need to pass some `Token` values as props `image, logoUri, name`

Then, we can simply do what I stated in the first comment https://github.com/gnosis/safe-react/pull/1169/files/28bf7a01dd3b1b3449fb521714a7f315d2b04f6c#r481123727

Reply:
Guys, while doing the refactor I found this code:
```
const CollectibleSelectField = ({ initialValue, tokens }) => {
  const classes = useCollectibleSelectFieldStyles()

  return (
    <Field
      className={classes.selectMenu}
      component={SelectField}
      disabled={!tokens.length}
      initialValue={initialValue}
      name="nftTokenId"
      renderValue={(nftTokenId) => <SelectedCollectible tokenId={nftTokenId} tokens={tokens} />}
      validate={required}
    >
      {tokens.map((token) => (
        <MenuItem key={`${token.assetAddress}-${token.tokenId}`} value={token.tokenId}>
          <ListItemIcon className={classes.tokenImage}>
            <TokenLogo height={28} tokenAddress={token.address} />
          </ListItemIcon>
          <ListItemText primary={token.name} secondary={`token ID: ${token.tokenId}`} />
        </MenuItem>
      ))}
    </Field>
  )
}
```

Please check this line: 
```
{tokens.map((token) => (
        <MenuItem key={`${token.assetAddress}-${token.tokenId}`} value={token.tokenId}>
          <ListItemIcon className={classes.tokenImage}>
            <TokenLogo height={28} tokenAddress={token.address} />
          </ListItemIcon>
          <ListItemText primary={token.name} secondary={`token ID: ${token.tokenId}`} />
        </MenuItem>
      ))}
```

We cannot use `useToken` hook there, we were just passing the address to the `TokenLogo` component and let him do it on his own, what do you suggest to do there?

@mikheevm @fernandomg

Reply:
@Agupane extract the code to a separate component and use the hook there

Reply:
Good idea, thanks!

Reply:
@mikheevm why wouldn't we just keep `TokenLogo` as it was before (with the hook inside), and move it away from the `src/components` directory?

With this approach, we will have a `TokenLogo` component that will only translate params.

If we think of `TokenLogo` as a component that will take care of everything for us, we can move it around, just pass it a token address and if today it uses a hook, tomorrow uses a REST service, later uses whatever, it will be transparent to the implementation.

Reply:
because the hook will take care of it. If we are displaying the token logo, most likely we want to display the symbol/name too, so the usage would be:

```
const token = useToken(address)

return (
  <>
    <Text>{token.symbol}</Text>
    <TokenLogo address={address} />
  </>
)

// TokenLogo

const { imageUrl } = useToken(address)

return <Image src={imageUrl} />
```

Do you see a problem there? I agree that the idea of having such component sounds nice, but IMO if you think more about it we don't really need it

Reply:
Hey guys I implemented that, could you take a look?